docs/changes/tests/13754.feature.rst

```diff
@@ -0,0 +1,3 @@
+Add the ``assert_type_is_type`` test helper function which asserts that types,
```

Comment:
Since this is exposed as "public" API here. Do we need a proper test to cover all the supported cases? The doctest snippet is pretty bare.